### PR TITLE
Feat/19 store token

### LINKED ISSUE #19: Higher order component for Login/JobApp conditional render
If token in storage valid = render JobApp
If token invalid = render Login

### PR DESCRIPTION
**Pull request by CamiloGarciaLaRotta**
### Description
This PR adds leverages`didMount` to check whether a `token` has been stored in the synced storage.
If so, it updates `state.token` with the value and renders the `JobApp` component
if not, it renders the `Login` component.

The storage, retrieval, and clearance of `Chrome.Storage.Sync` storage is handled by the `SyncStorage` module.

Note that #33 needs to be approved before this PR can be merged to keep a logical order.

Closes #19 

### Checklist
- [x] Code compiles correctly
- [x] All tests passing
- [x] There is an open issue linked to this PR
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation (if necessary)

### CODE REVIEW

**Comment by alexboii** [comment]
do you validate the token anywhere once you retrieve it?

**Comment by CamiloGarciaLaRotta** [comment]
Currently no, but It can be done by querying the `/self` auth endpoint.

Ince again, I think its outside of the scope of the PR. 
Let's get a complete UC by wiring up CV & JobApplication endpoints before putting more time into auth.

I'll create an issue for the validation of the token.

**Comment by alexboii** [comment]
it can't be outside of the scope of the PR

if you retrieve the token from storage, you need to check if it's still valid or not, it's all part of the same "operation" of login

**Review by alexboii — CHANGES_REQUESTED** [review]
needs token validation once retrieved from storage

**Comment by CamiloGarciaLaRotta** [comment]
ill leave it to you then

**Review by alexboii — APPROVED** [review]

**Comment by CamiloGarciaLaRotta** [comment]
LGTM

File: src/SyncStorage.re [hunk 1]
```diff
@@ -9,7 +9,7 @@ let refreshToken = maybeToken =>
   | Some(jwt) =>
     let token = Js.Dict.empty();
     Js.Dict.set(token, "token", Js.Json.string(jwt));
-    Sync.set(token, () => ());
+    Chrome.Extensions.Storage.Sync.set(token, () => ());
```
Comment (CamiloGarciaLaRotta) [inline src/SyncStorage.re]: Why calling the fn by its full name if there is already an `open Chrome.Extensions.Storage`

Comment (alexboii) [reply src/SyncStorage.re]: oh fuck I was just trying things and forgot to remove it

**Pull request merged**